tools/clang/lib/CodeGen/CGExprConstant.cpp

```diff
@@ -749,9 +812,62 @@ class ConstExprEmitter :
     case CK_HLSLCC_IntegralToFloating:
     case CK_HLSLCC_FloatingToIntegral:
     case CK_HLSLCC_FloatingToBoolean:
-      // Since these cast kinds have already been handled in ExprConstant.cpp,
-      // we can reuse the logic there.
-      return CGM.EmitConstantExpr(E, E->getType(), CGF);
```

Comment:
It's minor, but given how long the special case for matrices is, I think the code is more readable if it detects the common cast and returns at the top before delving into the matrix variant. instead of making readers scroll down to find the else